extensions/Push/lib/Connector/Phabricator.pm

```diff
@@ -59,9 +55,46 @@ sub should_send {
 }
 
 sub send {
+    my ( $self, $message ) = @_;
+
     my $logger = Bugzilla->push_ext->logger;
 
-    $logger->info('AUDIT');
+    my $data = $message->payload_decoded;
+    my $bug_data = $self->_get_bug_data($data) || return 0;
+    my $bug = Bugzilla::Bug->new( { id => $bug_data->{id}, cache => 1 } );
+
+    if(!is_public($bug)) {
+        $logger->info('Bailing on send because the bug is not public');
+        return 0;
+    }
+
+    my @attachments = grep {
+        $_->isobsolete == 0 &&
+        $_->contenttype eq PHAB_CONTENT_TYPE &&
+        $_->attacher->login eq 'phab-bot@bmo.tld'
+    } @{ $bug->attachments() };
+
+    if(@attachments){
+        my @rev_ids;
+        foreach my $attachment (@attachments) {
+            my ($rev_id) = ($attachment->filename =~ PHAB_ATTACHMENT_PATTERN);
+            next if !$rev_id;
+            push(@rev_ids, $rev_id);
+        }
+
+        if(@rev_ids) {
+            $logger->info('Getting info for revisions: ');
+            $logger->info(@rev_ids);
+
+            my @rev_details = get_revisions_by_ids(@rev_ids);
```

Comment:
Is it possible that, despite @rev_ids having `@` and pushing to it, that one item makes `@rev_ids` a string?  May explain the error in the Utils.pm file I mentioned above.